object_store/src/lib.rs

```diff
@@ -79,6 +86,22 @@
     doc = "* [`http`]: [HTTP/WebDAV Storage](https://datatracker.ietf.org/doc/html/rfc2518). See [`HttpBuilder`](http::HttpBuilder)"
 )]
 //!
+//! # Why not a Filesystem Interface?
+//!
+//! Whilst this crate does provide a [`BufReader`], the [`ObjectStore`] interface mirrors the APIs
+//! of object stores and not filesystems.
```

Comment:
I think it might help to give an example of what a filesystem API means. Perhaps something like

```suggestion
//! of object stores and not filesystems. For example, it purposely does not offer a `Seek` like API.
```